### PR TITLE
feat/python: More fixes for source

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Updates python code a bit more

## Links to any relevant issues

partial start on #1236 for the quick solves

### CODE REVIEW

File: bindings/python/examples/how_tos/nft_collection/01_mint_collection_nft.py [hunk 1]
```diff
@@ -43,12 +43,12 @@ def get_immutable_metadata(index: int, collection_id: str) -> str:
         "version": "v1.0",
         "type": "video/mp4",
         "uri": "ipfs://wrongcVm9fx47YXNTkhpMEYSxCD3Bqh7PJYr7eo5Ywrong",
-        "name": "Shimmer OG NFT #" + str(index),
+        "name": "Shimmer OG NFT #" +
+        str(index),
```
Comment (DaughterOfMars) [inline bindings/python/examples/how_tos/nft_collection/01_mint_collection_nft.py]: 🤨

File: bindings/python/iota_sdk/wallet/wallet.py [hunk 2]
```diff
@@ -18,7 +18,11 @@ class Wallet():
         handle: The wallet handle.
     """
 
-    def __init__(self, storage_path: Optional[str] = None, client_options: Optional[Dict[str, Any]] = None, coin_type: Optional[int] = None,
+    def __init__(self,
+                 storage_path: Optional[str] = None,
+                 client_options: Optional[Dict[str,
+                                               Any]] = None,
```
Comment (DaughterOfMars) [inline bindings/python/iota_sdk/wallet/wallet.py]: 🤨 🤨 🤨

File: bindings/python/tests/test_block.py
Comment (DaughterOfMars) [inline bindings/python/tests/test_block.py:1]: yeeeeesh

**Comment by kwek20** [comment]
its all autoformat :D dont blame me plz

File: bindings/python/iota_sdk/wallet/wallet.py (hunk 2 above)
Comment (kwek20) [reply bindings/python/iota_sdk/wallet/wallet.py]: undone this one

File: bindings/python/examples/how_tos/nft_collection/01_mint_collection_nft.py (hunk 1 above)
Comment (kwek20) [reply bindings/python/examples/how_tos/nft_collection/01_mint_collection_nft.py]: fixed

File: bindings/python/iota_sdk/wallet/wallet.py [hunk 3]
```diff
@@ -274,8 +277,7 @@ def destroy(self):
         return destroy_wallet(self.handle)
 
     @staticmethod
-    def __return_str_or_none(str):
-        if str:
-            return str
-        else:
-            return None
+    def __return_str_or_none(opt_str):
```
Comment (DaughterOfMars) [inline bindings/python/iota_sdk/wallet/wallet.py:280]: Why does this even exist?

Comment (kwek20) [reply bindings/python/iota_sdk/wallet/wallet.py:280]: didnt look into it

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflicts

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/python/examples/how_tos/account_output/governance_transition.py [hunk 4]
```diff
@@ -45,10 +45,10 @@
 
 
 def update_state_controller(unlock_condition):
+    """Update unlock condition if its a state controlled type."""
```
Comment (thibault-martinez) [inline bindings/python/examples/how_tos/account_output/governance_transition.py]:
```suggestion
    """Update unlock condition if it's a state controlled type."""
```

File: bindings/python/tests/test_block.py [hunk 5]
```diff
@@ -6,32 +6,231 @@
...
+                                                                                                                                                                                                                            "type": 3, "amount": "1000000", "unlockConditions": [
+                                                                                                                                                                                                                                {
+                                                                                                                                                                                                                                    "type": 0, "address": {
+                                                                                                                                                                                                                                        "type": 0, "pubKeyHash": "0x7ffec9e1233204d9c6dce6812b1539ee96af691ca2e4d9065daa85907d33e5d3"}}, {
```
Comment (thibault-martinez) [inline bindings/python/tests/test_block.py:101]: Can we check if formatting is somehow configurable? This is nonsense

Comment (kwek20) [reply bindings/python/tests/test_block.py:101]: ill do that in the issue when i address it fully

File: bindings/python/examples/.pylintrc [hunk 6]
```diff
@@ -6,6 +6,10 @@ disable=missing-module-docstring,
 	duplicate-code,
 	consider-using-with,
 	consider-using-f-string,
-	fixme
+	fixme, # TODOS 
+	too-many-instance-attributes,
+	too-many-arguments,
+	unused-argument # dict(locals()) doesnt work with this
```
Comment (thibault-martinez) [inline bindings/python/examples/.pylintrc]: Would it be too annoying to do it per case? It's a nice lint to loose :(

Comment (kwek20) [reply bindings/python/examples/.pylintrc]: fixed, but gh sucks again

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**